src/test/resources/com/puppycrawl/tools/checkstyle/grammar/InputRegressionJavaClass1Ast.txt

```diff
@@ -2219,12 +2219,12 @@ CLASS_DEF -> CLASS_DEF [17:0]
     |       |   |   `--RCURLY -> } [312:8]
     |       |   |--LITERAL_CATCH -> catch [312:10]
     |       |   |   |--LPAREN -> ( [312:16]
-    |       |   |   |--PARAMETER_DEF -> PARAMETER_DEF [312:29]
-    |       |   |   |   |--MODIFIERS -> MODIFIERS [312:29]
-    |       |   |   |   |--TYPE -> TYPE [312:29]
-    |       |   |   |   |   `--BOR -> | [312:29]
-    |       |   |   |   |       |--IDENT -> IOException [312:17]
-    |       |   |   |   |       `--IDENT -> ArrayIndexOutOfBoundsException [312:31]
+    |       |   |   |--PARAMETER_DEF -> PARAMETER_DEF [312:17]
+    |       |   |   |   |--MODIFIERS -> MODIFIERS [312:17]
+    |       |   |   |   |--TYPE -> TYPE [312:17]
+    |       |   |   |   |   |--IDENT -> IOException [312:17]
+    |       |   |   |   |   |--BOR -> | [312:29]
+    |       |   |   |   |   `--IDENT -> ArrayIndexOutOfBoundsException [312:31]
```

Comment:
please make catch with 3 of `|`.

Ideally AST should looks same as `a || b || c` in logical expressions [link](https://checkstyle.sourceforge.io/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LOR)
```
  |--EXPR -> EXPR
  |   `--LOR -> ||
  |       |--IDENT -> a
  |       `--IDENT -> b
```
But in logical expression priority of apply matters for  example `a && b || c`, it is a bit critical for logical expressions to be like this.

list of types in catch is more like https://checkstyle.sourceforge.io/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#TYPE_PARAMETERS
```
 |--TYPE_PARAMETERS -> TYPE_PARAMETERS
 |   |--GENERIC_START -> <
 |   |--TYPE_PARAMETER -> TYPE_PARAMETER
 |   |   `--IDENT -> A
 |   |--COMMA -> ,
 |   |--TYPE_PARAMETER -> TYPE_PARAMETER
 |   |   `--IDENT -> B
``` 

where separator is NOT a parent but sibling to elements.
Catch types are not bound by operator, they are list of types, so it is ok to make AST structure as you suggested.

ATTENTION: if you agree, please copy this explanation to issue description to exlpain rationale to users.
This issue is breaking compatibility.

Reply:
> please make catch with 3 of `|`.

https://github.com/checkstyle/checkstyle/pull/10136/files#r671704942

Reply:
I updated description